include/mqt-core/na/NAComputation.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "na/NADefinitions.hpp"
+#include "na/operations/NAOperation.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace na {
+class NAComputation {
+protected:
+  std::vector<std::shared_ptr<Point>> initialPositions;
```

Comment:
Similar to the other places where I annotated this, but this is likely more efficient when you just use `Point` instead of shared pointers.

Reply:
Yeah, but I need the pointers because I need to reference the Point from somewhere else because the concrete location of the point might be determined at a later stage only. To make it precise, when I construct a NAComputation and add an operation, I might not know the coordinates of the atom yet, so I add an undefined Point and when I decide where my atom should be in a later step, I update that point to the concrete location.

Reply:
The easy way around it is to duplicate NAComputation one more time and have one with shared_ptr and one without. I need some data structure during the mapping that offers this flexibility of not specifying the atoms position up front.

Reply:
Nah, guess that's not really worth it for now. Especially, if you only need the version with shared pointers.
Let's just stick with that for now and ignore my suggestions here.